plotly_resampler/figure_resampler/figurewidget_resampler.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+"""
+``FigureWidgetResampler`` wrapper around the plotly ``go.FigureWidget`` class.
+
+Utilizes the ``fig.layout.on_change`` method to enable dynamic resampling.
+
+"""
+
+from __future__ import annotations
+
+__author__ = "Jonas Van Der Donckt, Jeroen Van Der Donckt, Emiel Deprost"
+
+import re
+from typing import Tuple
+
+import plotly.graph_objects as go
+
+from .figure_resampler import AbstractFigureAggregator
+from ..aggregation import AbstractSeriesAggregator, EfficientLTTB
+
+
+class _FigureWidgetResamplerM(type(AbstractFigureAggregator), type(go.FigureWidget)):
+    # MetaClass for the FigureWidgetResampler
+    pass
+
+
+class FigureWidgetResampler(
+    AbstractFigureAggregator, go.FigureWidget, metaclass=_FigureWidgetResamplerM
+):
+    """Data aggregation functionality wrapper for ``go.FigureWidgets``.
+
+    .. attention::
+
+        * This wrapper only works within ``jupyter``-based environments.
+        * The ``.show()`` method returns a **static figure** on which the
+          **dynamic resampling cannot be performed**. To allow dynamic resampling,
+          you should just output the ``FigureWidgetResampler`` object in a cell.
+
+    """
+
+    def __init__(
+        self,
+        figure: go.FigureWidget | go.Figure = go.Figure(),
```

Comment:
Added `go.Figure()` as default, now it can  (just like the `FigureResampler`) be instantiated without arguments!

Reply:
Perhaps we should test this as well? :thinking:

Reply:
What do you mean? the FigureWidget tests already use `go.Figure`  as input :)